enhancements/olm/catalogd-query-endpoint.md

```diff
@@ -0,0 +1,585 @@
+   of the process is enforced by the linter CI job.
+
+See ../README.md for background behind these instructions.
+
+Start by filling out the header with the metadata for this enhancement.
+-->
+
+# catalogd-query-api
+
+## Summary
+
+Catalogd currently supports an HTTPS endpoint to access file-based catalog (FBC) contents
+but the endpoint provides complete catalog content in a single transaction.
+We would like to introduce a new HTTPS endpoint to provide more fine-grained access to 
+catalog contents. 
+
+<!--
+The `Summary` section is important for producing high quality
+user-focused documentation such as release notes or a development roadmap. It
+should be possible to collect this information before implementation begins in
+order to avoid requiring implementors to split their attention between writing
+release notes and implementing the feature itself.
+
+Your summary should be one paragraph long. More detail
+should go into the following sections.
+--> 
+
+## Motivation
+<!--
+This section is for explicitly listing the motivation, goals and non-goals of
+this proposal. Describe why the change is important and the benefits to users.
+-->
+
+The existing `/api/v1/all` endpoint returns the entire FBC, which the clients then need 
+to process in order to retrieve relevant information. This is very inefficient for 
+accesses which can be satisfied by a much smaller record set. 
+
+<!--
+redhat-operator-index:    55MB
+certified-operator-ind:   32MB
+community-operator-index: 29MB
+operatorhub.io:           21MB          <--- the benchmark 4-10s duration
+redhat-marketplace-index:  9MB
+-->
+
+Requiring clients to retrieve the full catalog can also result in a 4-10 second delay _per catalog_ even under optimal network conditions. 
+
+
+### User Stories
+
+* As a catalog client, I want to be able to pose space-efficient catalog queries
+
+### Goals
+
+* Provide an HTTPS endpoint which fulfills targeted catalog queries with a minimum record set.
+
+<!--
+Summarize the specific goals of the proposal. How will we know that
+this has succeeded?  A good goal describes something a user wants from
+their perspective, and does not include the implementation details
+from the proposal.
+-->
+
+### Non-Goals
+
+* Redesigning FBC schema to facilitate additional efficiencies.
+* Implementing adoption by existing tools (for e.g. kubectl plugins).
+* Response scaling beyond ~1024 simultaneous request.
+* Implementing API discovery mechanisms.
+
+
+<!--
+What is out of scope for this proposal? Listing non-goals helps to
+focus discussion and make progress. Highlight anything that is being
+deferred to a later phase of implementation that may call for its own
+enhancement.
+-->
+
+## Proposal
+
+This proposal introduces an additional HTTPS endpoint to an existing catalogd API.  The existing HTTPS "all" endpoint will remain as a default option; the user will be able to enable this new capability via a feature gate.
```

Comment:
Should we talk about how we plan to deprecate "all" once the new endpoint is GA?